False positive `unspecified-encoding` for Path objects opened in binary mode

### Bug description

```python
'''
This snippet raises no warnings in previous versions, but raises unspecified-encoding in 2.11.0
Opening in binary mode -- either 'wb' or 'rb'.
'''
import pathlib
fp = pathlib.Path(__file__)
with fp.open('rb') as f:
    pass
```


### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
************* Module a
a.py:6:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```


### Expected behavior

none

### Pylint version

```shell
pylint 2.11.0
astroid 2.8.0
Python 3.9.6 (v3.9.6:db3ff76da1, Jun 28 2021, 11:49:53) 
[Clang 6.0 (clang-600.0.57)]
```


### OS / Environment

macOS

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -20,6 +20,9 @@ Release date: TBA
 ..
   Put bug fixes that should not wait for a new minor version here
 
+* ``unspecified-encoding`` now checks the encoding of ``pathlib.Path()`` correctly
+
+  Closes #5017
 
 
 What's New in Pylint 2.11.0?

--- a/pylint/checkers/stdlib.py
+++ b/pylint/checkers/stdlib.py
@@ -531,7 +531,7 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
                     or isinstance(node.func, nodes.Attribute)
                     and node.func.attrname in OPEN_FILES_ENCODING
                 ):
-                    self._check_open_encoded(node)
+                    self._check_open_encoded(node, inferred.root().name)
             elif inferred.root().name == UNITTEST_CASE:
                 self._check_redundant_assert(node, inferred)
             elif isinstance(inferred, nodes.ClassDef):
@@ -609,11 +609,18 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
             ):
                 self.add_message("bad-open-mode", node=node, args=mode_arg.value)
 
-    def _check_open_encoded(self, node: nodes.Call) -> None:
+    def _check_open_encoded(self, node: nodes.Call, open_module: str) -> None:
         """Check that the encoded argument of an open call is valid."""
         mode_arg = None
         try:
-            mode_arg = utils.get_argument_from_call(node, position=1, keyword="mode")
+            if open_module == "_io":
+                mode_arg = utils.get_argument_from_call(
+                    node, position=1, keyword="mode"
+                )
+            elif open_module == "pathlib":
+                mode_arg = utils.get_argument_from_call(
+                    node, position=0, keyword="mode"
+                )
         except utils.NoSuchArgumentError:
             pass
 

--- a/tests/functional/u/unspecified_encoding_py38.py
+++ b/tests/functional/u/unspecified_encoding_py38.py
@@ -72,3 +72,11 @@ LOCALE_ENCODING = None
 Path(FILENAME).write_text("string")  # [unspecified-encoding]
 Path(FILENAME).write_text("string", encoding=None)  # [unspecified-encoding]
 Path(FILENAME).write_text("string", encoding=LOCALE_ENCODING)  # [unspecified-encoding]
+
+LOCALE_ENCODING = locale.getlocale()[1]
+Path(FILENAME).open("w+b")
+Path(FILENAME).open()  # [unspecified-encoding]
+Path(FILENAME).open("wt")  # [unspecified-encoding]
+Path(FILENAME).open("w+")  # [unspecified-encoding]
+Path(FILENAME).open("w", encoding=None)  # [unspecified-encoding]
+Path(FILENAME).open("w", encoding=LOCALE_ENCODING)

--- a/tests/functional/u/unspecified_encoding_py38.txt
+++ b/tests/functional/u/unspecified_encoding_py38.txt
@@ -19,3 +19,7 @@ unspecified-encoding:65:0::Using open without explicitly specifying an encoding:
 unspecified-encoding:72:0::Using open without explicitly specifying an encoding:HIGH
 unspecified-encoding:73:0::Using open without explicitly specifying an encoding:HIGH
 unspecified-encoding:74:0::Using open without explicitly specifying an encoding:HIGH
+unspecified-encoding:78:0::Using open without explicitly specifying an encoding:HIGH
+unspecified-encoding:79:0::Using open without explicitly specifying an encoding:HIGH
+unspecified-encoding:80:0::Using open without explicitly specifying an encoding:HIGH
+unspecified-encoding:81:0::Using open without explicitly specifying an encoding:HIGH

